presto-main/src/main/java/com/facebook/presto/sql/gen/DereferenceCodeGenerator.java

```diff
@@ -41,50 +43,67 @@ public BytecodeNode generateExpression(BytecodeGeneratorContext generator, Type
         checkArgument(arguments.size() == 2);
         CallSiteBinder callSiteBinder = generator.getCallSiteBinder();
 
-        BytecodeBlock block = new BytecodeBlock().comment("DEREFERENCE").setDescription("DEREFERENCE");
-        Variable wasNull = generator.wasNull();
-        Variable rowBlock = generator.getScope().createTempVariable(Block.class);
-        int index = ((Number) ((ConstantExpression) arguments.get(1)).getValue()).intValue();
+        // Collect all nested intermediateDerefs
+        Stack<RowExpression> nestedDerefs = new Stack<RowExpression>();
```

Comment:
- `nestedDereferences`
- also, we don't use `Stack` data structure. Try something different like `Deque`

Reply:
Done.
Changed to List/ArrayList